### PR TITLE
feat: add short table output

### LINKED ISSUE #933: Output table does not fit in my terminal
# Describe the bug
I am running `kubescape scan -v` and the printed table is to wide for my termianl

# Environment
OS: Mac, M1
Version: the version that Kubescape reports when you run `kubescape version`
```
Your current version is: v2.0.176
```

# Steps To Reproduce
run `kubescape scan -v`

# Expected behavior
A nice and readable table 

# Actual Behavior
A non-readable table 

# Additional context
The table should be rendered so it will match the default terminal width of 80 columns

### PR DESCRIPTION
**Pull request by XDRAGON2002**
## Overview

As mentioned in the issue, the extremely wide tabular output doesn't fit in smaller terminals in a representable manner, hence a smaller more compact tabular format has been added which automatically infers based on the given terminal if the wider output needs to be rendered or the smaller.

## Additional Information

In order to detect the terminal size a new dependency has been used, and automatically displays the smaller output tables if the terminal width is not able to support the original table output.
It does so by pivoting the table hence decreasing the width but instead increasing the height. A tradeoff made depending on the available width.

## How to Test

On a smaller terminal screen with width less than 120, run any of:
1. `kubescape list controls`
2. `kubescape scan resource.yaml`
3. `kubescape scan resource.yaml -v`

## Examples/Screenshots

Before:

image
image
image

After:

image
image
image

## Related issues/PRs:

Resolved #933 

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by XDRAGON2002** [comment]
@craigbox I've added the ability to automatically detect the terminal width, and formatted the table as discussed.
Do let me know if there are any further changes needed! 
Thanks!

File: core/pkg/resultshandling/printer/v2/utils.go [hunk 1]
```diff
@@ -82,3 +86,12 @@ func finalizeResources(results []resourcesresults.Result, allResources map[strin
...
 }
+
+func CheckShortTerminalWidth() bool {
+	width, _, err := term.GetSize(int(os.Stdin.Fd()))
+	if err != nil {
+		// Default to larger output table
+		return false
+	}
+	return width <= 120
+}
```
Comment (craigbox) [inline core/pkg/resultshandling/printer/v2/utils.go:97]: I had hoped we could calculate the width the table is going to be, and then use that as input to this function: if the table can be displayed within the existing width, do so, else go to short mode.

Comment (XDRAGON2002) [reply core/pkg/resultshandling/printer/v2/utils.go:97]: A problem that I had felt with that approach was the fact that we'd have to figure out the width of the largest row to determine this, which might lead to some redundancy, but on second thought, I guess there might be a way if we slightly alter the order in which we construct the table, I'll take a shot with the approach and update the results here.

**Comment by XDRAGON2002** [comment]
@craigbox I've made the required changes and now the table format is decided based on whether the large output tables are able to be displayed on the current terminal or not, if not then the smaller table is displayed ;)

**Comment by craigbox** [comment]
wonderful. to @dwertent & @matthyx for review please!

**Comment by craigbox** [comment]
Can you please fix the conflicts, and we can get this merged soon I hope.

**Comment by dwertent** [comment]
@XDRAGON2002  I see there are still conflicts.
Probably because we merged #1320 .

**Comment by dwertent** [comment]
@XDRAGON2002  [Here is the issue](https://github.com/kubescape/kubescape/actions/runs/5809307982/job/15747841347?pr=1292#step:11:393).

I recommend you run before you push the following commands:
```
go test "-tags=static,gitenabled" -v ./...
```
and 
```
make build
```

**Comment by dwertent** [comment]
It works as expected, I will merge it as soon as the tests pass :)

**Comment by XDRAGON2002** [comment]
@dwertent this one is good to go now.

**Pull request merged**